tcms/core/tests/test_ajax.py

```diff
@@ -91,3 +97,89 @@ def test_update_default_tester_non_existing_user(self):
         self.assertEqual(result['response'], 'Default tester not found!')
 
         self._assert_default_tester_is(None)
+
+
+class Test_Tag(BasePlanCase):
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.url = reverse('ajax-tags')
+        cls.test_tag = TestTagFactory()
+        cls.test_tag_two = TestTagFactory()
+        cls.test_plan = TestPlanFactory()
+        cls.test_plan_two = TestPlanFactory()
+        cls.test_case = TestCaseFactory()
+        cls.test_run = TestRunFactory()
+
+    def test_add_tag_to_test_plan(self):
+        response = self.client.get(self.url, {
+            'tags': self.test_tag,
+            'plan': self.test_plan.plan_id,
+            'a': 'add'
+        })
+
+        self.assertEqual(response.status_code, http.client.OK)
+        self.assertEqual(self.test_plan.tag.count(), 1)
+        self.assertTrue(self.test_tag in self.test_plan.tag.all())
+
+    def test_add_tag_to_test_case(self):
+        response = self.client.get(self.url, {
+            'tags': self.test_tag,
+            'case': self.test_case.case_id,
+            'a': 'add'
+        })
+
+        self.assertEqual(response.status_code, http.client.OK)
+        self.assertEqual(self.test_case.tag.count(), 1)
+        self.assertTrue(self.test_tag in self.test_case.tag.all())
+
+    def test_add_tag_to_test_run(self):
+        response = self.client.get(self.url, {
+            'tags': self.test_tag,
+            'run': self.test_run.run_id,
+            'a': 'add'
+        })
+
+        self.assertEqual(response.status_code, http.client.OK)
+        self.assertEqual(self.test_run.tag.count(), 1)
+        self.assertTrue(self.test_tag in self.test_run.tag.all())
+
+    def test_add_tag_to_multiple_plans(self):
+        """ Add one tag to multiple test plans """
+
+        response = self.client.get(self.url, {
+            'tags': self.test_tag,
+            'plan': [self.test_plan.plan_id, self.test_plan_two.plan_id],
+            'a': 'add',
+            't': 'json'
+        })
+
+        self.assertEqual(response.status_code, http.client.OK)
+        self.assertEqual(response.content, b'{"response": "ok"}')
+        self.assertEqual(self.test_plan.tag.count(), 1)
+        self.assertEqual(self.test_plan_two.tag.all().count(), 1)
+        self.assertTrue(self.test_tag in self.test_plan.tag.all() and
+                        self.test_tag in self.test_plan_two.tag.all())
+
+    def test_add_tag_to_multiple_plans_serialized(self):
+        """
+        Add one tag to multiple test plans
+
+        Because of the 'f': 'serialized' flag we expect
+        the response to contain the plans for which the tag has been added
+        """
+        response = self.client.get(self.url, {
+            'tags': self.test_tag,
+            'plan': [self.test_plan.plan_id, self.test_plan_two.plan_id],
+            'a': 'add',
+            't': 'json',
+            'f': 'serialized'
```

Comment:
From what I can tell in the JavaScript files and in the templates there isn't a single place which sends a request with the serialized flag enabled. You should investigate this into more detail (search for /management/tags/ and id_tag_form). 

If this turns out to be the case (needs to be very carefully verified) then corresponding piece of code and tests should be removed.